src/ucp/core/ucp_context.c

```diff
@@ -639,6 +642,28 @@ static ucs_status_t ucp_fill_resources(ucp_context_h context,
          * client-server connection establishment via sockaddr, don't use it */
         if ((num_tl_resources > 0) ||
             (context->tl_mds[md_index].attr.cap.flags & UCT_MD_FLAG_SOCKADDR)) {
+            /*List of memory type MDs*/
+            if (context->tl_mds[md_index].attr.cap.mem_type != UCT_MD_MEM_TYPE_HOST) {
+                /* check if mem type MD is already in the list */
+                for (j = 0; j < context->num_mem_type_mds; j++) {
+                    if (context->mem_type_tl_mds[j]->attr.cap.mem_type ==
+                        context->tl_mds[md_index].attr.cap.mem_type) {
+                        break;
+                    }
+                }
+                if (j == context->num_mem_type_mds) {
+                    context->mem_type_tl_mds = ucs_realloc(context->mem_type_tl_mds,
```

Comment:
suggestion: can just allocate an array which can potentially hold all mds, instead of reallocating

Reply:
fixed. allocated #UCT_MD_MEM_TYPE_LAST size as we are not storing the duplicates mds